accounts/checkers.go

```diff
@@ -0,0 +1,615 @@
+		r *lnrpc.SendResponse) (proto.Message, error) {
+
+		status := lnrpc.Payment_IN_FLIGHT
+		if len(r.PaymentError) > 0 {
+			status = lnrpc.Payment_FAILED
+		}
+
+		hash, err := lntypes.MakeHash(r.PaymentHash)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing payment hash: %v",
+				err)
+		}
+
+		route := r.PaymentRoute
+		totalAmount := int64(0)
+		if route != nil {
+			totalAmount = route.TotalAmtMsat + route.TotalFeesMsat
+		}
+
+		return checkSendResponse(
+			ctx, service, status, hash, totalAmount,
+		)
+	}
+
+	checkers := CheckerMap{
+		// Invoices:
+		"/lnrpc.Lightning/AddInvoice": mid.NewResponseRewriter(
+			&lnrpc.Invoice{},
+			&lnrpc.AddInvoiceResponse{},
+			func(ctx context.Context,
+				t *lnrpc.AddInvoiceResponse) (proto.Message,
+				error) {
+
+				hash, err := lntypes.MakeHash(t.RHash)
+				if err != nil {
+					return nil, fmt.Errorf("error parsing "+
+						"invoice hash: %v", err)
+				}
+
+				acct, err := AccountFromContext(ctx)
+				if err != nil {
+					return nil, err
+				}
+
+				return nil, service.AssociateInvoice(
+					acct.ID, hash,
+				)
+			}, mid.PassThroughErrorHandler,
+		),
+		"/lnrpc.Lightning/ListInvoices": mid.NewResponseRewriter(
+			&lnrpc.ListInvoiceRequest{},
+			&lnrpc.ListInvoiceResponse{},
+			func(ctx context.Context,
+				t *lnrpc.ListInvoiceResponse) (proto.Message,
+				error) {
+
+				filteredInvoices, err := filterInvoices(ctx, t)
+				if err != nil {
+					return nil, fmt.Errorf("error "+
+						"filtering invoices: %v", err)
+				}
+
+				t.Invoices = filteredInvoices
+				return t, nil
+			}, mid.PassThroughErrorHandler,
+		),
+		"/lnrpc.Lightning/LookupInvoice": mid.NewRequestChecker(
+			&lnrpc.PaymentHash{},
+			&lnrpc.Invoice{},
+			func(ctx context.Context, t *lnrpc.PaymentHash) error {
+				acct, err := AccountFromContext(ctx)
+				if err != nil {
+					return err
+				}
+
+				hash, err := lntypes.MakeHash(t.RHash)
+				if err != nil {
+					return fmt.Errorf("error parsing "+
+						"payment hash: %v", err)
+				}
+
+				if _, ok := acct.Invoices[hash]; !ok {
+					return fmt.Errorf("invoice does not " +
+						"belong to this account")
+				}
+
+				return nil
+			},
+		),
+
+		// Payments:
+		"/lnrpc.Lightning/SendPayment": mid.NewFullChecker(
+			&lnrpc.SendRequest{},
+			&lnrpc.SendResponse{},
+			func(ctx context.Context, r *lnrpc.SendRequest) error {
+				return checkSend(
+					ctx, chainParams, service, r.Amt,
+					r.AmtMsat, r.PaymentRequest, r.FeeLimit,
+				)
+			}, sendResponseHandler, mid.PassThroughErrorHandler,
+		),
+		"/lnrpc.Lightning/SendPaymentSync": mid.NewFullChecker(
+			&lnrpc.SendRequest{},
+			&lnrpc.SendResponse{},
+			func(ctx context.Context, r *lnrpc.SendRequest) error {
+				return checkSend(
+					ctx, chainParams, service, r.Amt,
+					r.AmtMsat, r.PaymentRequest, r.FeeLimit,
+				)
+			}, sendResponseHandler, mid.PassThroughErrorHandler,
+		),
+		// routerrpc.Router/SendPayment is deprecated.
```

Comment:
Is it? 🧐

Reply:
Yes, since there's the `routerrpc.Router/SendPaymentV2` which should be used. We still allow the `lnrpc.Lightning/SendPayment` to be used, even though it is also kind of old. Maybe you meant that one?